resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/server/servlet/HttpServletResponseHeaders.java

```diff
@@ -150,13 +150,32 @@ public void putAll(Map<? extends String, ? extends List<Object>> map)
       for (Map.Entry<? extends String, ? extends List<Object>> entry : map.entrySet())
       {
          List<Object> objs = entry.getValue();
-         for (Object obj : objs)
-         {
-            add(entry.getKey(), obj);
+         // When the header key does not exist, undertow creates it and saves this value.
+         // When the header key exists undertow clears the existing values and saves this value.
+         // All subsequent values must be added so they will be appended properly.
+         putResponseHeader(entry.getKey(), objs.get(0)); // rls test
```

Comment:
I tried making this adjustment.  This change breaks the return headers count.  Each object in the header input list is registered as an unique array element in undertow.  Submitting our list as a single
string breaks the expected header count upon return of headers.  Parsing the string back into a list on
our side does not seen like a good idea to me.
@ronsigal @asoldano